com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -807,5 +817,71 @@ static void SetupPerFrameShaderConstants()
             // Used when subtractive mode is selected
             Shader.SetGlobalVector(PerFrameBuffer._SubtractiveShadowColor, CoreUtils.ConvertSRGBToActiveColorSpace(RenderSettings.subtractiveShadowColor));
         }
+
+#if ADAPTIVE_PERFORMANCE_2_1_0_OR_NEWER
+        static void ApplyAdaptivePerformance(ref CameraData cameraData)
+        {
+            var noFrontToBackOpaqueFlags = SortingCriteria.SortingLayer | SortingCriteria.RenderQueue | SortingCriteria.OptimizeStateChanges | SortingCriteria.CanvasOrder;
+            if (AdaptivePerformance.AdaptivePerformanceRenderSettings.SkipFrontToBackSorting)
+                cameraData.defaultOpaqueSortFlags = noFrontToBackOpaqueFlags;
+
+            var MaxShadowDistanceMultiplier = AdaptivePerformance.AdaptivePerformanceRenderSettings.MaxShadowDistanceMultiplier;
+            cameraData.maxShadowDistance *= MaxShadowDistanceMultiplier;
+
+            var RenderScaleMultiplier = AdaptivePerformance.AdaptivePerformanceRenderSettings.RenderScaleMultiplier;
+            cameraData.renderScale *= RenderScaleMultiplier;
+
+            // TODO
+            if (!cameraData.isStereoEnabled)
+            {
+                cameraData.cameraTargetDescriptor.width = (int)(cameraData.camera.pixelWidth * cameraData.renderScale);
+                cameraData.cameraTargetDescriptor.height = (int)(cameraData.camera.pixelHeight * cameraData.renderScale);
```

Comment:
We have to be careful with render scaling, as this kind of RT scale recreates render textures. The render scale in URP is meant to be done once to select your game resolution (720p, 1080p, 1440p... )

for something dynamic as adaptive performance is probably better to have a different scaling system.

Reply:
Good to know! For GLES I'd have though that might be the case, it this also on Vulcan? It's currently used as fall-back if dynamic resolution would not be available.

Reply:
I'd suggest to not use this render scale with adaptive performance, it will possibly allocate textures per-frame and cause issues.